arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/popup/PopupViewModel.kt

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2021 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.popup
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.esri.arcgisruntime.ArcGISRuntimeException
+import com.esri.arcgisruntime.concurrent.ListenableFuture
+import com.esri.arcgisruntime.data.Feature
+import com.esri.arcgisruntime.data.FeatureEditResult
+import com.esri.arcgisruntime.data.FeatureTable
+import com.esri.arcgisruntime.data.ServiceFeatureTable
+import com.esri.arcgisruntime.mapping.popup.Popup
+import com.esri.arcgisruntime.mapping.popup.PopupManager
+import com.esri.arcgisruntime.toolkit.util.Event
+import com.esri.arcgisruntime.toolkit.util.raiseEvent
+
+/**
+ * The view model that represents a Popup. It supports:
+ *
+ * - Viewing a GeoElement's attributes
+ * - Editing a GeoElement's attributes as well as saving
+ * - Deleting a GeoElement
+ *
+ * A PopupViewModel can be bound to a [PopupView] for visualisation of the GeoElement's
+ * attributes and editing experience.
```

Comment:
```suggestion
 * attributes and editing experience.
 * For an example of how to use it by binding it to a PopupView see [PopupViewTestActivity]. 
```

Reply:
PopupViewTestActivity is in the toolkit-test-app module and since it is not a dependency for the arcgis-android-toolkit module, we cannot successfully add the link to PopupViewTestActivity. 
I have however added this comment without the [PopupViewTestActivity] link

Reply:
Ok, that makes sense 👍